serenity/metrics_helper.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef SERENITY_METRICS_HELPER_HPP
+#define SERENITY_METRICS_HELPER_HPP
+
+#include <stout/error.hpp>
+#include <stout/try.hpp>
+
+#include "mesos/mesos.hpp"
+
+namespace mesos {
+namespace serenity {
+
+inline Try<double_t> CountCpuUsage(const ResourceUsage_Executor& previous,
+                            const ResourceUsage_Executor& current) {
+  if (!current.has_statistics() ||
+      !previous.has_statistics() ||
+      !current.statistics().has_timestamp() ||
+      !current.statistics().has_cpus_user_time_secs() ||
+      !current.statistics().has_cpus_system_time_secs() ||
+      !previous.statistics().has_timestamp() ||
+      !previous.statistics().has_cpus_user_time_secs() ||
+      !previous.statistics().has_cpus_system_time_secs()) {
+    return Error("Cannot count CPU usage, Parameter does not have required "
+                     "statistics");
+  }
+
+  double_t samplingDuration = current.statistics().timestamp() -
+                              previous.statistics().timestamp();
+  double_t cpuTimeUsage = (current.statistics().cpus_system_time_secs() +
+                           current.statistics().cpus_user_time_secs()) -
+                          (previous.statistics().cpus_system_time_secs() +
+                           previous.statistics().cpus_user_time_secs());
+
+  return cpuTimeUsage / samplingDuration;
+}
+
+inline Try<double_t> CountIpc(const ResourceUsage_Executor& previous,
+                       const ResourceUsage_Executor& current) {
+  if (!current.has_statistics() || !previous.has_statistics())
+    return Error("Cannot count IPC, Parameter does not have required "
+                     "statistics");
+  if (!current.statistics().has_perf() ||
+      !previous.statistics().has_perf() ||
+      !current.statistics().perf().has_timestamp() ||
+      !previous.statistics().perf().has_timestamp() ||
+      !current.statistics().perf().has_cycles() ||
+      !previous.statistics().perf().has_cycles() ||
+      !current.statistics().perf().has_instructions() ||
+      !previous.statistics().perf().has_instructions()) {
+    return Error("Cannot count IPC usage, Parameter does not have required "
+                     "perf statistics");
+  }
+
+  double_t instructions = current.statistics().perf().instructions() -
```

Comment:
From Estimator / Controller point of view it's difficult to find out if given ResourceUsage_Executor is gathered from continously running job or from another job with re-used freshly destroyed job's ExecutorID. The only way to do it is to hope that between destroying and creating action we will fetch ResourceUsage without that executor (: Is this what you had in mind?